src/Container.ts

```diff
@@ -40,38 +49,42 @@ export class Container {
     /**
      * Registers a new service.
      *
-     * @param name Service name. Optional
-     * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param descriptor
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(descriptor: ServiceDescriptor) {
+        this.registeredServices.push(descriptor);
     }
 
     /**
      * Retrieves the service with the specific name or given type from the service container.
-     * Optionally parameters can be pass in the case if instance is initialized in the container for the first time.
+     * Optionally, parameters can be passed in case if instance is initialized in the container for the first time.
      */
     static get<T>(type: ConstructorFunction<T>, params?: any[]): T;
     static get<T>(name: string, params?: any[]): T;
     static get<T>(typeOrName: ConstructorFunction<T>|string, params?: any[]): T {
 
         // normalize parameters
-        let type: Function, name: string;
+        let type: Function;
+        let name: string;
+        let factory: Function;
         if (typeof typeOrName === "string") {
             name = <string> typeOrName;
         } else {
             type = <Function> typeOrName;
         }
 
         // find if service was already registered
-        const registeredService = this.findRegisteredService(name, type);
-        // console.log("registeredService: ", registeredService);
-        if (registeredService) {
-            if (!type)
-                type = registeredService.type;
-            if (!params)
-                params = registeredService.params;
+        const serviceDescriptor = this.findRegisteredService(name, type);
+        if (serviceDescriptor) {
+            if (!type) {
+                type = serviceDescriptor.type;
+            }
+            if (!params) {
+                params = serviceDescriptor.params;
+            }
+            if (serviceDescriptor.factory) {
+                factory = serviceDescriptor.factory;
+            }
```

Comment:
`in the industry` where? in c++ or php?). Styles guides are different, some of them popular some of them less popular. In php styleguide with always-{} is more popular. In javascript I don't think its more popular. I always was against {} in simple single line statements because it makes code uglier and more complicated. Simply need to know when you to put {} in a single line statements and when not, and format with above and below newlines properly.